kolibri/plugins/learn/assets/src/views/ChannelRenderer/ContentItem.vue

```diff
@@ -159,21 +161,20 @@
       setWasIncomplete() {
         this.wasIncomplete = this.progress < 1;
       },
+      cacheProgress() {
+        setContentNodeProgress({ content_id: this.content.content_id, progress: this.progress });
+      },
       updateInteraction({ progress, interaction }) {
         this.updateContentSession({
           interaction,
           progress,
-        }).then(() => updateContentNodeProgress(this.contentNodeId, this.progress));
+        }).then(this.cacheProgress);
       },
       updateProgress(progress) {
-        this.updateContentSession({ progress }).then(() =>
-          updateContentNodeProgress(this.contentNodeId, this.progress)
-        );
+        this.updateContentSession({ progress }).then(this.cacheProgress);
```

Comment:
I don't understand why the content session is being updated and _then_ the progress is cached?

Reply:
The previous utility function was slightly erroneously named, but it was updating the client side cache of content node progress. This is doing the same just with a different mechanism.

Reply:
Added a code comment to the `this.cacheProgress` method to try to explain this better.